agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/AiSampler.java

```diff
@@ -112,9 +125,10 @@ private static SamplingResult useLocalParentDecisionIfPossible(Context parentCon
       return SamplingResult.drop();
     }
     if (parentSpan instanceof ReadableSpan) {
-      Long itemCount = ((ReadableSpan) parentSpan).getAttribute(AiSemanticAttributes.ITEM_COUNT);
-      if (itemCount != null) {
-        return new RecordAndSampleWithItemCount(itemCount);
+      Double sampleRate =
```

Comment:
```suggestion
      Double parentSampleRate =
```

At first, I have not noticed it was the sample rate from the parent span